docs/proposals/nvidia-gpu-enhancement.md

```diff
@@ -0,0 +1,64 @@
+# Add NVIDIA GPU discovery and assignment capabilities to Kubelet.
+
+## Background:
+
+So far we only support 1 GPU(```/dev/nvidia0```) and the GPU device need to be configured as a Kubelet's parameter manually. But in production environment, there could have multiple GPUs(```/dev/nvidia0, /dev/nvidia1, /dev/nvidia2 ...```) installed on 1 Minion, also 1 micro service could needs more than 1 GPU, and there could have thousands of Minions which means if Kubelet could discover GPU by itself would be great.
+
+
+## Goal:
+
+ - Have the capabilities of GPU discovery to let Kubelet could aware more than 1 NVIDIA GPU and make the deployment easier.
```

Comment:
I did nothing special for virtual GPUs, they do not work unless they act like the physical GPUs. I think it's better we consider it after manage GPUs by NVML in the future. Because if we do it at that time, even the virtual GPU acts like physical GPU, the NVML still can tell the different.